MantidQt/API/src/QwtWorkspaceSpectrumData.cpp

```diff
@@ -52,21 +52,15 @@ QwtWorkspaceSpectrumData *QwtWorkspaceSpectrumData::copyWithNewSource(
 
 /** Size of the data set
  */
-size_t QwtWorkspaceSpectrumData::size() const {
-  if (m_binCentres || m_isHistogram) {
-    return m_Y.size();
-  }
-
-  return m_X.size();
-}
+size_t QwtWorkspaceSpectrumData::size() const { return m_hist.size(); }
 
 /**
 Return the x value of data point i
 @param i :: Index
 @return x X value of data point i
 */
 double QwtWorkspaceSpectrumData::getX(size_t i) const {
-  return m_binCentres ? (m_X[i] + m_X[i + 1]) / 2 : m_X[i];
+  return m_binCentres ? m_hist.points()[i] : m_hist.binEdges()[i];
```

Comment:
Is there a way of converting a  whole histogram from bin edges to points in place? I'm aware that you can use `convertToCounts` and `convertToFrequencies` for Y and E, but is there any equivalent for X?